section_start_rules.js

```diff
@@ -274,72 +275,76 @@ function processRule(rule, section_types) {
 
 // Constructor for section start rules
 function Rule(key, values_hash, rule_number, section_types) {
-  // get a rule count as string
-  if (rule_number < 9) {
-    criteria_count = '0' + rule_number;
-  } else {
-    criteria_count = rule_number;
-  }
+  // skip any Section Start entries without criteria
+  if (values_hash.hasOwnProperty('contiguous_block_criteria_01') || rule_number > 1) {
+    // get a rule count as string
+    if (rule_number < 9) {
+      criteria_count = '0' + rule_number;
+    } else {
+      criteria_count = rule_number;
+    }
 
-  // set values for 'rule'
-  this.rule_name = key + "_" + criteria_count;
-  this.ss_name = key;
-  if (values_hash.hasOwnProperty('section_required')) {
-    this.section_required = values_hash['section_required']['value']
-    this.insert_before = values_hash['section_required']['insert_before'];
-  } else {
-    this.section_required = false;
-    this.insert_before = [];
-  }
-  if (values_hash.hasOwnProperty('position')) {
-    this.position = values_hash['position'];
-  } else {
-    this.position = '';
-  }
-  this.multiple = values_hash["contiguous_block_criteria_" + criteria_count]['multiple'];
-  this.styles = values_hash["contiguous_block_criteria_" + criteria_count]['styles'];
-  if (values_hash["contiguous_block_criteria_" + criteria_count].hasOwnProperty('optional_heading_styles')) {
-    this.optional_heading_styles = values_hash["contiguous_block_criteria_" + criteria_count]['optional_heading_styles'];
-  } else {
-    this.optional_heading_styles = [];
-  }
-  if (values_hash["contiguous_block_criteria_" + criteria_count].hasOwnProperty('first_child')) {
-    this.first_child = true;
-    this.first_child_text = values_hash["contiguous_block_criteria_" + criteria_count]['first_child']['text'];
-    this.first_child_match = values_hash["contiguous_block_criteria_" + criteria_count]['first_child']['match'];
-  } else {
-    this.first_child = false;
-    this.first_child_text = [];
-    this.first_child_match = '';
-  }
-  this.required_styles = values_hash["contiguous_block_criteria_" + criteria_count]['previous_sibling']['required_styles'];
-  if (values_hash["contiguous_block_criteria_" + criteria_count].hasOwnProperty('previous_until')) {
-    this.previous_until = values_hash["contiguous_block_criteria_" + criteria_count]['previous_until']
-  } else {
-    this.previous_until = [];
-  }
+    // set values for 'rule'
```

Comment:
Can you add some more commenting throughout this block to walk through what each of these steps is doing? Try to think about if a brand new programmer is opening up this script for the first time, what background info do they need to know in order to make updates to this script?